planner/core/logical_plan_builder.go

```diff
@@ -1560,13 +1572,27 @@ func (b *PlanBuilder) resolveGbyExprs(p LogicalPlan, gby *ast.GroupByClause, fie
 	}
 	for _, item := range gby.Items {
 		resolver.inExpr = false
-		retExpr, _ := item.Expr.Accept(resolver)
+		var retExpr ast.Node
+		inPrepare := (len(b.ctx.GetSessionVars().PreparedParams) == 0)
+		if param, ok := item.Expr.(*driver.ParamMarkerExpr); ok && !inPrepare {
```

Comment:
Will there be one case that `!ok && inPrepare` or `ok && inPrepare` is true?

Reply:
Thank you for your detailed comments. The condition should be fixed. The updated PR will address the issue.